author_api/models.py

```diff
@@ -30,9 +58,80 @@ def host(self):
     def url(self):
       return self.host + 'author/' + str(self.id)
 
+    objects = AuthorManager()
+
+    # mixing controller logic with model logic unfortunately
+    def add_follower(self, follower):
+        """Create a follower from an Author/CachedAuthor model"""
+        _cached = CachedAuthor.objects.get(id = follower.id)
+
+        # Prevent duplicate entries
+        if not self.followers.filter(id = follower.id):
+            self.followers.add(_cached)
+
+        # Bidirectional follow relationships become friendships
+        try:
+            newfriend = Author.objects.get(id = follower.id)
+
+            # Only create a friendship if self is being followed back
+            if newfriend.followers.filter(id = self.id):
+                newfriend.add_friend(self)
+                self.add_friend(newfriend)
+        except:
+            # This will fail if:
+            #   1. The follower is actually foreign hosted
+            # TODO
+            # To determine if we can add friendship, must query foreign node
+            # self.add_friend(newfriend)
+            pass
+
+    # Call add_follower instead
+    def add_friend(self, follower):
```

Comment:
Also, for friend related functions, aka this one and removeFriend, the "follower" argument should be "friend".